int32 type
---
int32 error